BrainPortal/lib/boutiques_input_copier.rb

```diff
@@ -0,0 +1,223 @@
+
+#
+# CBRAIN Project
+#
+# Copyright (C) 2008-2024
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+#
+# Some tools modify the cached version of the input directory.
+#
+# This can lead to some issues when the input is used multiple times
+# on the same Bourreau and the cached version is already modified.
+#
+# To solve this issue, we can create a copy of the directory in the working
+# directory and use this copy in the command line.
+#
+#     apptool [DIRECTORY_INPUT] ...
+#
+# with the following input in Boutiques:
+#
+#     {
+#       "description": "A directory that contain the input data...",
+#       "id": "input",
+#       "name": "Input",
+#       "optional": true,
+#       "type": "File",
+#       "command-line-flag": "--input",
+#       "value-key": "[INPUT]"
+#     },
+#
+# and in the `cbrain:integrator_modules` section look like:
+#
+#     "BoutiquesInputCopier": {
+#       "parent_input_id": {"default": boolean, "hide": boolean, "description": string},
+#      },
+#
+# The final command line will be:
+#
+#     apptool [INPUT_COPY]...
+#
+#
+module BoutiquesInputCopier
+
+  # Note: to access the revision info of the module,
+  # you need to access the constant directly, the
+  # object method revision_info() won't work.
+  Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+
+  ############################################
+  # Portal Side Modifications
+  ############################################
+
+  # Adjust the description for the input with the mention of
+  # the fake parent directory information
+  #
+  # For each input in BoutiquesInputCopier section create a new entry in the descriptor
+  # that will be a flag to choose if the input will be copied if the checkbox is selected
+  def descriptor_for_before_form #:nodoc:
```

Comment:
Change this to a new method with an argument (descriptor) and then the other methods that you want to override will call it as needed. Don't make framework methods call each other, each shoudl have their own distinct path through the class hierarchy without cross-over.

```ruby
def extension_of_descriptor(descriptor)
   blah blah
end

def descriptor_for_before_form
   desc = super
   extension_of_descriptor(descriptor)
end

def descriptor_for_form
   desc = super
   extension_of_descriptor(descriptor)
end

def descriptor_for_after_form
   desc = super
   extension_of_descriptor(descriptor)
end
```